### ISSUE TITLE
Don't add a selector to services of types ExternalName

### ISSUE THREAD

**Issue by benjamin-bergia** [kubernetes-sigs/kustomize#2294]
I have a service of type ExternalName. In my kustomization file I specified a list of commonLabels.
Currently kustomize add the commonLabels to all objects metadata and service selectors. This is expected and correct. Unfortunately it also add a selector to the services of type ExternalName. Kubernetes doesn't seem to complain about it but the behaviour of a service of type ExternalName that also has a selector seems to be to entirely bypass the externalName and act as a normal service.

Would it be possible to add a check to the commonLabels transformer to add labels only in the metadata of the ExternalName services and don't add any selector.

**Comment by benjamin-bergia** [comment]
I think this is also messing up headless services.

**Comment by nbn** [comment]
I believe I am encountering this problem too. Is there a way to bypass it in anyway?

As an example, having this kustomization file:

```
commonLabels:
  owner: me

resources:
  - service.yaml
```

And a service:

````
apiVersion: v1
kind: Service
metadata:
  name: myservice
spec:
  selector:
    podlabel: mypod
````
Give me the label as metadata, but also in the selector, like this:

```
apiVersion: v1
kind: Service
metadata:
  labels:
    owner: me
  name: myservice
spec:
  selector:
    owner: me
    podlabel: mypod
```

My problem is that not all pods are managed by this kustomization project, so I cannot label as usual.
When I use [kapp](http://get-kapp.io), i can disable label-scoping (as they call it) by annotating the service.
```
metadata:
  annotations:
    kapp.k14s.io/disable-label-scoping: ""
```

Can something similar be expressed in kustomize?

**Comment by nbn** [comment]
I think this is very similar to issue #2034

**Comment by benjamin-bergia** [comment]
The only way at the moment is to avoid commonLabels for this service. Which is hard in our case since we have multiple level of kustomization files

**Comment by benjamin-bergia** [comment]
I am hitting this again. Just having a 
```yaml
commonLabels:
  app.kubernetes.io/name: myapp
```
for example, will make it impossible to use endpoints that are manually defined. Since:
> For headless Services that define selectors, the endpoints controller creates Endpoints records in the API, and modifies the DNS configuration to return records (addresses) that point directly to the Pods backing the Service.
Without selectors

> For headless Services that do not define selectors, the endpoints controller does not create Endpoints records. However, the DNS system looks for and configures either:
> - CNAME records for ExternalName-type Services.
> - A records for any Endpoints that share a name with the Service, for all other type

[Doc on services](https://kubernetes.io/docs/concepts/services-networking/service/#headless-services)

**Comment by xvzf** [comment]
/reopen
this is still occuring

**Comment by tringuyen-yw** [comment]
Still happening 2022-05-19. This is quite a serious bug. Why is it closed?

```
kubectl version --short

Client Version: v1.24.0
Kustomize Version: v4.5.4
Server Version: v1.22.6
WARNING: version difference between client (1.24) and server (1.22) exceeds the supported minor version skew of +/-1
```

**Comment by m3nax** [comment]
/reopen
this is still occuring

```
kubectl version --short

Client Version: v1.23.6
Server Version: v1.23.6-2+2a84a218e3cd52
```

**Comment by qixiaobo** [comment]
![image](https://user-images.githubusercontent.com/2881751/209546781-ec22f4e0-9475-4948-bba8-b572ce975c5f.png)
![image](https://user-images.githubusercontent.com/2881751/209546798-3ecc32f0-6251-4b28-9e9d-7fb914ec1de6.png)

**Comment by sass1997** [comment]
In my opinion this needs to be fixed since a service of type ExternalName will never work with selectors

**Comment by kouk** [comment]
Apparently the [corresponding bug in Kubernetes](https://github.com/kubernetes/kubernetes/issues/105986
) was [fixed in 1.27.0](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.27.md#v1270 )

**Comment by jonenst** [comment]
@benjamin-bergia  can you reopen ? While it is possible in some cases to workaround by creating the externalname services after the transformer runs, this feature would make the labels: transform hugely easier to use, remove a big footgun.
